### PR TITLE
Parallelize IndexBGEN

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: hail/src/main/scala/is/hail/io/bgen/IndexBgen.scala [hunk 1]
```diff
@@ -57,52 +57,51 @@ object IndexBgen {
...
-          .sortWith { case ((k1, _), (k2, _)) => keyOrdering.lt(k1, k2) }
-
         val partIdx = TaskContext.get.partitionId()
 
-        using(new IndexWriter(sHadoopConfBc.value.value, indexFilePaths(partIdx), keyType, annotationType, attributes = attributes)) { iw =>
-          keys.foreach { case (k, offset) =>
-            iw += (k, offset, Row())
+        using(new IndexWriter(sHadoopConfBc.value.value, indexFilePaths(partIdx), indexKeyType, annotationType, attributes = attributes)) { iw =>
+          it.foreach { r =>
+            iw += (r.deleteFields(Array(offsetIdx, fileIdxIdx)), r.getLong(offsetIdx), Row())
```
Comment (cseed) [inline hail/src/main/scala/is/hail/io/bgen/IndexBgen.scala]: Drop the `deleteFields`.  This has no benefit over constructing a new row with the fields you want: `Row(r(offsetIdx), r(fileIdx))`.  I'll read the rest more closely, but high level, nicely done!

Comment (cseed) [reply hail/src/main/scala/is/hail/io/bgen/IndexBgen.scala]: Oops, I mean `Row(r(locusIdx), r(allelesIdx))`, I guess.

Comment (cseed) [reply hail/src/main/scala/is/hail/io/bgen/IndexBgen.scala]: Also, you might assert `r.getInt(fileIdxIdx) == partIdx`.

**Comment by cseed** [comment]
@jigold is this ready for another look?

**Comment by jigold** [comment]
Yes, but I was waiting to do the timing experiments until I got a JAR and ZIP from the CI. It's been pending every time I look.

**Pull request merged**